app/tests/integration/nimbus/conftest.py

```diff
@@ -124,9 +125,12 @@ def _create_experiment(selenium, home_page, data):
 def perform_kinto_action():
     def _perform_kinto_action(selenium, base_url, action):
         selenium.get("http://kinto:8888/v1/admin")
-        kinto_login = Login(selenium, base_url).wait_for_page_to_load()
-        kinto_login.kinto_auth.click()
-        kinto_dashbard = kinto_login.login()
+        try:
+            kinto_login = Login(selenium, base_url).wait_for_page_to_load()
+            kinto_login.kinto_auth.click()
+            kinto_dashbard = kinto_login.login()
+        except NoSuchElementException:
+            kinto_dashbard = Dashboard(selenium, base_url).wait_for_page_to_load()
```

Comment:
What's this doing?
Handling a case where we're already current logged into the dashboard?

Reply:
Exactly that!!